python_modules/dagster/dagster_tests/core_tests/structured_config_tests/test_errors.py

```diff
@@ -0,0 +1,87 @@
+import pytest
+from dagster import Config, op
+from dagster._config.structured_config import ConfigurableResource
+from dagster._core.errors import DagsterInvalidPythonicConfigDefinitionError
+
+
+def test_invalid_config_type_basic() -> None:
+    class MyUnsupportedType:
+        pass
+
+    class DoSomethingConfig(Config):
+        unsupported_param: MyUnsupportedType
+
+    with pytest.raises(
+        DagsterInvalidPythonicConfigDefinitionError,
+        match="""Error defining Dagster config class <class 'test_errors.test_invalid_config_type_basic.<locals>.DoSomethingConfig'> on field 'unsupported_param'.
+Unable to resolve config type <class 'test_errors.test_invalid_config_type_basic.<locals>.MyUnsupportedType'>.
+
+
+This value can be a:
+    - Python primitive type
+        - int, float, bool, str, list
+    - A Python Dict or List type containing other valid types
+    - Custom data classes extending dagster.Config
+    - A Pydantic discriminated union type""",
```

Comment:
I dont know what a "Pydantic discriminated union type" is. Is there more detail that can be added here or a documentation page we can link to?